### PR TITLE
contrib/{echo,gorilla}: add AppSec response-status test

### PR DESCRIPTION
**Pull request by Hellzy**
- Add test to check response status with AppSec rules
- Rework echo.v4 AppSec entry point to fix a bug uncovered by this test

### CODE REVIEW

File: contrib/gorilla/mux/mux_test.go [hunk 1]
```diff
@@ -384,4 +384,25 @@ func TestAppSec(t *testing.T) {
...
+		require.NoError(t, err)
+		require.Equal(t, 404, res.StatusCode)
+
+		finished := mt.FinishedSpans()
+		require.Len(t, finished, 1)
+		event := finished[0].Tag("_dd.appsec.json").(string)
+		require.NotNil(t, event)
+		require.True(t, strings.Contains(event, "server.response.status"))
+		require.True(t, strings.Contains(event, "nfd-000-001"))
+
```
Comment (Julio-Guerra) [inline contrib/gorilla/mux/mux_test.go:406]:
```suggestion
```

File: contrib/labstack/echo.v4/echotrace.go [hunk 2]
```diff
@@ -57,6 +55,12 @@ func Middleware(opts ...Option) echo.MiddlewareFunc {
 			// pass the span through the request context
 			c.SetRequest(request.WithContext(ctx))
 			// serve the request to the next middleware
+
```
Comment (Julio-Guerra) [inline contrib/labstack/echo.v4/echotrace.go:58]:
```suggestion
```

File: contrib/labstack/echo.v4/echotrace.go [hunk 3]
```diff
@@ -57,6 +55,12 @@ func Middleware(opts ...Option) echo.MiddlewareFunc {
...
+			if appsecEnabled {
+				afterMiddleware := useAppSec(c, span)
+				defer afterMiddleware()
+			}
+
```
Comment (Julio-Guerra) [inline contrib/labstack/echo.v4/echotrace.go:63]:
```suggestion
```

File: contrib/labstack/echo.v4/echotrace_test.go [hunk 4]
```diff
@@ -367,5 +367,26 @@ func TestAppSec(t *testing.T) {
...
+			require.NoError(t, err)
+			require.Equal(t, 404, res.StatusCode)
+
+			finished := mt.FinishedSpans()
+			require.Len(t, finished, 1)
+			event := finished[0].Tag("_dd.appsec.json").(string)
+			require.NotNil(t, event)
+			require.True(t, strings.Contains(event, "server.response.status"))
+			require.True(t, strings.Contains(event, "nfd-000-001"))
+
```
Comment (Julio-Guerra) [inline contrib/labstack/echo.v4/echotrace_test.go:389]:
```suggestion
```

**Review by Julio-Guerra — DISMISSED** [review]
Good catch. I added some nitpicks on empty lines.
I also think that we should put all those tests in a common test package next time we add more so they get simpler to maintain.

File: contrib/gorilla/mux/mux_test.go (hunk 1 above)
Comment (Hellzy) [reply contrib/gorilla/mux/mux_test.go:406]: > Good catch. I added some nitpicks on empty lines. I also think that we should put all those tests in a common test package next time we add more so they get simpler to maintain.

Created a [JIRA ticket](https://datadoghq.atlassian.net/browse/APPSEC-3515?atlOrigin=eyJpIjoiNzFlY2YzOTllNzNmNDQyZThjYTI2MjVmOGE0ODE2YTIiLCJwIjoiaiJ9) to track this.

**Comment by Hellzy** [comment]
@knusbaum @ajgajg1134 could one of you have a look at these changes?

**Review by Julio-Guerra — APPROVED** [review]

**Review by knusbaum — APPROVED** [review]

**Pull request merged**